minio/signer.py

```diff
@@ -96,6 +96,9 @@ def presign_v4(method, url, access_key, secret_key, region=None,
 
     headers_to_sign = dict(headers)
 
+    if response_headers is not None:
+        headers_to_sign.update(response_headers)
+
```

Comment:
If `response_headers` is an empty `dict`, you could do simply `headers_to_sign.update(response_headers)`